typescript/testsuite/testsuite_impl/basic_datastore_and_api_test/basic_datastore_and_api_test_.ts

```diff
@@ -0,0 +1,238 @@
+
+//TODO TODO TODO (Ali) - review these helper methods after making final changes to network_impl
+async function getDatastoreServiceConfigurations(): Promise<[ContainerCreationConfig, (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => Result<ContainerRunConfig, Error>]> {
+	const datastoreContainerCreationConfig: ContainerCreationConfig = await getDataStoreContainerCreationConfig();
+
+	const datastoreRunConfigFunc: (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => Result<ContainerRunConfig, Error> = await getDataStoreRunConfigFunc();
+	return [datastoreContainerCreationConfig, datastoreRunConfigFunc];
+}
+
+async function getDataStoreContainerCreationConfig(): Promise<ContainerCreationConfig> {
+	const containerCreationConfig: ContainerCreationConfig = new ContainerCreationConfigBuilder( //TODO (Ali) - may need to make ContainerCreationConfig async
+		DATASTORE_IMAGE,
+	).withUsedPorts(
+		new Set(""+DATASTORE_PORT+"/tcp"),
+	).build()
+	return containerCreationConfig;
+}
+
+async function getDataStoreRunConfigFunc(): Promise<(ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => Result<ContainerRunConfig, Error>> {
+	const runConfigFunc: (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => Result<ContainerRunConfig, Error> = 
+	(ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => {
+		return ok(new ContainerRunConfigBuilder().build());
+	}
+	return runConfigFunc;
+}
+
+async function getApiServiceConfigurations(datastoreClient: DatastoreClient): Promise<[ContainerCreationConfig, (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => Result<ContainerRunConfig, Error>]> {
+	const configInitializingFunc: (fp: number) => Promise<Result<null, Error>> = await getApiServiceConfigInitializingFunc(datastoreClient);
+
+	const apiServiceContainerCreationConfig: ContainerCreationConfig = getApiServiceContainerCreationConfig(configInitializingFunc);
+
+	const apiServiceGenerateRunConfigFunc: (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => Result<ContainerRunConfig, Error> = await getApiServiceRunConfigFunc();
+	return [apiServiceContainerCreationConfig, apiServiceGenerateRunConfigFunc];
+}
+
+async function getApiServiceConfigInitializingFunc(datastoreClient: DatastoreClient): Promise<(fp: number) => Promise<Result<null, Error>>> { //Note: Making simplification that file descriptor is just number
+	const configInitializingFunc: (fp: number) => Promise<Result<null, Error>> = async (fp: number) => { //TOOD (Ali) - might require changes in ConfigRunFactory in kurt-client due to async
```

Comment:
So this guy _should_ be `async`, and this is correct

Reply:
PS: there should be a return type assertion specified in the `async (fp: number)`